python/src/iceberg/transforms.py

```diff
@@ -0,0 +1,446 @@
+            )
+        else:
+            self._apply = lambda v: self._diff_func(
+                datetime.utcfromtimestamp(v / 1000000)
+            )
+
+        self._result_type = DateType if self._name == "day" else IntegerType
+
+    def apply(self, value: int) -> int:
+        return self._apply(value)
+
+    def can_transform(self, target: Type) -> bool:
+        if self._type == DateType:
+            return target == DateType
+        else:  # self._type is either TimestampType or TimestamptzType
+            return target == TimestampType or target == TimestamptzType
+
+    def result_type(self, source_type: Type) -> Type:
+        return self._result_type
+
+    def preserves_order(self) -> bool:
+        return True
+
+    def satisfies_order_of(self, other: Transform) -> bool:
+        if self == other:
+            return True
+
+        if isinstance(other, Time):
+            return (
+                Time._TIME_SATISFIED_ORDER[self._name]
+                <= Time._TIME_SATISFIED_ORDER[other._name]
+            )
+
+        return False
+
+    def dedup_name(self) -> str:
+        return "time"
+
+
+class Identity(Transform):
+    _HUMAN_STRING_MAP = {
+        DateType: lambda v: transform_util.human_day(v),
+        TimeType: lambda v: transform_util.human_time(v),
+        TimestampType: lambda v: transform_util.human_timestamp(v),
+        TimestamptzType: lambda v: transform_util.human_timestamptz(v),
+        BinaryType: lambda v: transform_util.base64encode(v),
+    }
+
+    def __init__(self, source_type: Type):
+        if isinstance(source_type, FixedType):
+            to_human_string = transform_util.base64encode
+        else:
+            to_human_string = Identity._HUMAN_STRING_MAP.get(
+                source_type, transform_util.to_string
+            )
+        super().__init__(
+            "identity",
+            f"transforms.identity(source_type={repr(source_type)})",
+            to_human_string,
+        )
+        self._type = source_type
+
+    def apply(self, value):
+        return value
+
+    def can_transform(self, target: Type) -> bool:
+        return target.is_primitive
+
+    def preserves_order(self) -> bool:
+        return True
+
+    def satisfies_order_of(self, other: Transform) -> bool:
+        return other.preserves_order()
+
+
+class Truncate(Transform):
+    _VALID_TYPES = {IntegerType, LongType, StringType, BinaryType}
+
+    def __init__(self, source_type: Type, width: int):
+        if source_type not in Truncate._VALID_TYPES and not isinstance(
+            source_type, DecimalType
+        ):
+            raise ValueError(f"Cannot truncate type: {source_type}")
+
+        super().__init__(
+            f"truncate[{width}]",
+            f"transforms.truncate(source_type={repr(source_type)}, width={width})",
+            transform_util.base64encode
+            if source_type == BinaryType
+            else transform_util.to_string,
```

Comment:
Can you move this inline expression? It makes the code hard to read.

Reply:
👌  create a method for that.